bokeh/resources.py

```diff
@@ -36,78 +42,193 @@
 from .core.templates import CSS_RESOURCES, JS_RESOURCES
 from .model import Model
 from .settings import settings
-from .util.paths import bokehjsdir
+from .util.paths import ROOT_DIR, bokehjsdir
 from .util.session_id import generate_session_id
 
-#-----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
 # Globals and constants
-#-----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
 
 DEFAULT_SERVER_HOST = "localhost"
 DEFAULT_SERVER_PORT = 5006
 DEFAULT_SERVER_HTTP_URL = "http://%s:%d/" % (DEFAULT_SERVER_HOST, DEFAULT_SERVER_PORT)
 
 # __all__ defined at the bottom on the class module
 
-#-----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
 # General API
-#-----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
 
-#-----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
 # Dev API
-#-----------------------------------------------------------------------------
+# -----------------------------------------------------------------------------
+
+
+_SRI_HASHES = None
+
+
+def get_all_sri_hashes():
+    """ Report SRI script hashes for all versions of BokehJS.
+
+    Bokeh provides `Subresource Integrity`_ hashes for all JavaScript files that
+    are published to CDN for full releases. This function returns a dictionary
+    that maps version strings to sub-dictionaries that JavaScipt filenames to
+    their hashes.
+
+    Returns:
+        dict
+
+    Example:
+
+        The returned dict will map version strings to sub-dictionaries for each
+        version:
+
+        .. code-block:: python
+
+            {
+                '1.4.0': {
+                    'bokeh-1.4.0.js': 'vn/jmieHiN+ST+GOXzRU9AFfxsBp8gaJ/wvrzTQGpIKMsdIcyn6U1TYtvzjYztkN',
+                    'bokeh-1.4.0.min.js': 'mdMpUZqu5U0cV1pLU9Ap/3jthtPth7yWSJTu1ayRgk95qqjLewIkjntQDQDQA5cZ',
+                    ...
+                }
+                '1.3.4': {
+                    ...
+                }
+                ...
+            }
+
+    .. _Subresource Integrity: https://developer.mozilla.org/en-US/docs/Web/Security/Subresource_Integrity
+
+    """
+    global _SRI_HASHES
+
+    if not _SRI_HASHES:
+        _SRI_HASHES = json.load(open(join(ROOT_DIR, "_sri.json")))
+
+    return dict(_SRI_HASHES)
```

Comment:
FYI this is behind a function:

* so that a copy can be supplied
* it takes ~150us to load this on my machine. A few things like that start to add up at import time. I'd rather defer this cost until use.